launch/launch/actions/declare_launch_argument.py

```diff
@@ -79,19 +81,41 @@ def __init__(
         name: Text,
         *,
         default_value: Optional[SomeSubstitutionsType] = None,
-        description: Text = 'no description given',
+        description: Optional[Text] = None,
+        choices: List[Text] = None,
         **kwargs
     ) -> None:
         """Create a DeclareLaunchArgument action."""
         super().__init__(**kwargs)
         self.__name = name
+        self.__logger = launch.logging.get_logger(__name__)
         if default_value is None:
             self.__default_value = default_value
         else:
             self.__default_value = normalize_to_list_of_substitutions(default_value)
-        self.__description = description
+        if choices is not None:
+            # Check if a non substitution default value is provided and is a valid choice
+            if default_value is not None and not isinstance(default_value, Substitution):
+                if default_value not in choices:
+                    self.__logger.error(
+                        "Provided default_value '{}' is not in provided choices '{}'.".format(
+                            default_value, choices)
+                    )
+                    raise RuntimeError(
+                        "Provided default_value '{}' is not in provided choices '{}'.".format(
+                            default_value, choices))
+
+        if description is None:
+            if choices is None:
+                self.__description = 'no description given'
+            else:
+                self.__description = 'One of: ' + str(choices)
+        else:
+            self.__description = description
+            if choices is not None:
+                self.__description += " Valid choices are: " + str(choices)
 
-        self.__logger = launch.logging.get_logger(__name__)
+        self.__choices = choices
```

Comment:
@v-lopez meta: shall we check that `choices` is not an empty list as well?

Reply:
Sure, please check https://github.com/ros2/launch/pull/483/commits/20797ba791c818e6ae86d2f30f6f642e9933d90c.